.github/actions/conditional/conditional.sh

```diff
@@ -1,54 +1,86 @@
 #!/bin/bash -e
 
-REMOTE=$1
-BASE_REF=$2
+REMOTE="$1"
+BASE_REF="$2"
 
-if [ "$BASE_REF" != "" ]; then
-  if [ "$GITHUB_OUTPUT" != "" ]; then
-    echo "--------------------------------------------------------------------------------"
-    echo "Fetching '$BASE_REF' in '`git remote get-url $REMOTE`'"
+CONDITIONS_FILE=".github/actions/conditional/conditions"
+
+[ "$BASE_REF" != "" ] && IS_PR=true || IS_PR=false
+[ "$GITHUB_OUTPUT" != "" ] && IS_GITHUB_ACTIONS=true || IS_GITHUB_ACTIONS=false
+
+if [ "$IS_PR" == true ]; then
+  # Fetch remote if running on GitHub Actions
+  if [ "$IS_GITHUB_ACTIONS" == true ]; then
+    echo "========================================================================================"
+    echo "Fetching '$BASE_REF' in '$(git remote get-url "$REMOTE")'"
     echo "--------------------------------------------------------------------------------"
-    git fetch --depth 1 $REMOTE $BASE_REF
+    git fetch --depth 1 "$REMOTE" "$BASE_REF"
   fi
 
-  echo "--------------------------------------------------------------------------------"
-  echo "Changes compared to '$BASE_REF' in '`git remote get-url $REMOTE`'"
-  echo "--------------------------------------------------------------------------------"
-  git diff $REMOTE/$BASE_REF --name-only
+  # Get list of changes files
+  echo "========================================================================================"
+  echo "Changes compared to '$BASE_REF' in '$(git remote get-url "$REMOTE")'"
+  echo "----------------------------------------------------------------------------------------"
+  CHANGED_FILES=$(git diff "$REMOTE/$BASE_REF" --name-only)
+  echo "$CHANGED_FILES"
+fi
+
+echo "========================================================================================"
+if [ "$IS_PR" == true ]; then
+  echo "Matching regex"
+  echo "----------------------------------------------------------------------------------------"
 else
-  echo "--------------------------------------------------------------------------------"
   echo "Not a pull request, marking everything as changed"
 fi
 
-echo "--------------------------------------------------------------------------------"
-echo "Run conditions"
-echo "--------------------------------------------------------------------------------"
+declare -A JOB_CONDITIONS
+
+readarray -t CONDITIONS <<< "$(cat "$CONDITIONS_FILE" | grep -v '^[ ]*#' | grep -v '^[ ]*$')"
+
+for C in "${CONDITIONS[@]}"; do
+  read -r -a CONDITION <<< "$C"
 
-cat .github/actions/conditional/conditions | grep '=' | grep -v '#' | while read c; do
-  KEY=`echo $c | cut -d '=' -f 1`
-  PATTERN=`echo $c | cut -d '=' -f 2`
+  if [ "$IS_PR" == true ]; then
+    PATTERN="${CONDITION[0]}"
 
-  if [ "$BASE_REF" != "" ]; then
-    DIFF=`echo $PATTERN | xargs git diff $REMOTE/$BASE_REF --name-only`
-    if [ "$DIFF" != "" ]; then
-      CHANGED=true
+    # Convert pattern to regex
+    REGEX=$(echo "$PATTERN" | sed 's|\.|[.]|g' | sed 's|/$|/.*|g' | sed 's|^*|.*|g')
```

Comment:
This only covers simple pattern cases but fails for e.g. `*/src/test/*.java` which is translated to
`.*/src/test/*[.]java`
while it should be
`.*/src/test/.*[.]java`

I would incline to simply use regexes directly in the `conditions` file rather than translating. I leave to your discretion whether you'd like to follow this suggestion.

Reply:
I didn't want to use regex in the conditions file itself as it makes it hard to maintain. We can tune/extend on it if needed based on what patterns we want to support. It's mostly at the moment trying to support what's supported in CODEOWNERS, to have some consistency cross the two.